### ISSUE TITLE
Generated link breed primitives are not being highlighted

### ISSUE THREAD

**Issue by qiemem** [NetLogo/NetLogo#1131]
Example:

![image](https://cloud.githubusercontent.com/assets/28215/17668962/61849130-62d1-11e6-8179-91cc7ddef9d9.png)

The `my-[out/in]-<breed>` primitives are similarly not highlighted.

**Comment by mrerrormessage** [comment]
Yep, this is a good enhancement. If there's a PR this can go into hexy, otherwise we'll put it in the next release.

**Comment by qiemem** [comment]
Oh! I thought this was a regression.

**Comment by mrerrormessage** [comment]
Nope. I was pretty sure it wasn't because highlighting doesn't necessarily rely on a compiling version of the code (and then I checked 5.2, which didn't have it). If the code doesn't compile, it's very hard to determine what breed names are. (Yes, I know if you just count tokens and assume braces, etc. but all of that is a lot of work to highlight relatively few, seldom-used primitives).

**Comment by Idloj** [comment]
Does this issue apply to turtle breeds as well?

**Comment by mrerrormessage** [comment]
Yes, this covers both turtle and link breeds. While this can certainly be done for compiling code, I'm not certain whether this can be accomplished for non-compiling code (unless we have a "best-effort" error-repairing compiler).

**Comment by Idloj** [comment]
I think the forgiving-parser @Akarshit made for jump-to-declaration can be evolved to a more complete forgiving-parser that can be used in other places (including this issue of course).

btw, I believe the title should be changed
